aether-kernel/aether/kernel/api/entity_extractor.py

```diff
@@ -379,11 +383,43 @@ def put_nested(_dict, keys, value):
             _dict[keys[0]] = put_nested(_dict[keys[0]], keys[1:], value)
         except KeyError:  # Level doesn't exist yet
             _dict[keys[0]] = put_nested({}, keys[1:], value)
-    else:
+    # Key has an array accessor like child[0]
+    # Array accessors are only allowed at the last path element
+    elif '[' in keys[0]:
```

Comment:
wouldn't it be better `if ARRAY_ACCESSOR_REGEX.match(keys[0]) is not None:`?

Reply:
The regex is greedy and only splits on `[` & `]` so anything with <2 matches isn't an array accessor.